src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginUpdateHeader.java

```diff
@@ -0,0 +1,117 @@
+package com.github.manolo8.darkbot.gui.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.plugins.PluginHandler;
+import com.github.manolo8.darkbot.extensions.plugins.PluginUpdater;
+import com.github.manolo8.darkbot.gui.components.MainButton;
+import com.github.manolo8.darkbot.utils.I18n;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class PluginUpdateHeader extends JPanel {
+
+    private static final MessageFormat titleFormatter = new MessageFormat(I18n.get("plugins.config_button.update_header"));
+    private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("EEE, d MMM, hh:mm a", Locale.ROOT);
+    private String status;
+
+    private final PluginUpdater pluginUpdater;
+    private final PluginHandler pluginHandler;
+
+    private final Title title;
+    private final JProgressBar progressBar;
+    private final UpdateAllButton updateAllButton;
+    private final CheckUpdateButton checkUpdateButton;
+
+    PluginUpdateHeader(Main main) {
+        super(new MigLayout("ins 0, gap 0, fill", "[grow][][]", "[][grow]"));
+
+        this.pluginUpdater = main.pluginUpdater;
+        this.pluginHandler = main.pluginHandler;
+        this.status = !pluginUpdater.hasAnyUpdates() ? I18n.get("plugins.config_button.up_to_date") + " ," : "";
+        this.title = new Title();
+        this.progressBar = new JProgressBar();
+        this.updateAllButton = new UpdateAllButton();
+        this.checkUpdateButton = new CheckUpdateButton();
+
+        progressBar.setVisible(false);
+        progressBar.setBorderPainted(false);
+
+        add(title);
+        add(updateAllButton);
+        add(checkUpdateButton);
+        add(progressBar, "dock south, spanx");
```

Comment:
```suggestion
        add(this.title = new Title());
        add(this.updateAllButton = new UpdateAllButton());
        add(this.checkUpdateButton = new CheckUpdateButton());
        add(this.progressBar = new JProgressBar(), "dock south, spanx");
        
        progressBar.setBorderPainted(false);
        
        refreshUI();
```

You're repeating some code from here and the refreshUI method, you can simplify it a bit